### PR TITLE
Full sync operations: timer and non blocking

### PR DESCRIPTION
**Pull request by nukky94**
The goal of this PR is to use built-in timer logger from finatra and provide a non blocking timeoutable and cancellable futures for sync operations

### CODE REVIEW

File: src/main/scala/co/ledger/wallet/daemon/services/PoolsService.scala [hunk 1]
```diff
@@ -69,14 +71,33 @@ class PoolsService @Inject()(daemonCache: DaemonCache) extends DaemonService {
...
       } yield accounts
-      val resultFuture = accountsFuture.map { accounts =>
-        accounts.map {
-          case (poolName, walletName, a) =>
-            val f = a.sync(poolName, walletName)
-            Try(Await.result(f, 30.minute)).recoverWith{ case t =>
-              Failure(AccountSyncException(poolName, walletName, a.getIndex, t))
+
+      // custom ec to avoid thread starvation in the global ec
+      val fullSyncEc: ExecutionContext = ExecutionContext.fromExecutor(
```
Comment (luchenyuxx) [inline src/main/scala/co/ledger/wallet/daemon/services/PoolsService.scala]: not sure about the purpose of this ec. the sync won't happen in global ec, they are directed to lib core ec.

File: src/main/scala/co/ledger/wallet/daemon/utils/FutureUtils.scala [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  /**
+    * Scala Future doesn't provide a cancel function.
+    * As a workaround, you could use the firstCompletedOf to wrap 2 futures.
+    * The future you want to cancel and a future that comes from a custom Promise.
+    * You could then cancel the thus created future by failing the promise.
+    * @param future Caller passes a future to execute
+    * @param onError Error handler
+    * @return Future[T]
+    */
+  def cancellable[T](future: Future[T])(onError: => Unit)(implicit ec: ExecutionContext): (() => Unit, Future[T]) = {
```
Comment (luchenyuxx) [inline src/main/scala/co/ledger/wallet/daemon/utils/FutureUtils.scala]: Just not that the `future` is not cancelled after the `cancellable` is called. It's an improper name

File: src/main/scala/co/ledger/wallet/daemon/services/PoolsService.scala (hunk 1 above)
Comment (nukky94) [reply src/main/scala/co/ledger/wallet/daemon/services/PoolsService.scala]: The goal here is to use a different ec for timeoutable operation:
https://imgur.com/a/AbMJZpH
But if we look at it closer, it still uses the global ec.
Should I correct that or revert it?

File: src/main/scala/co/ledger/wallet/daemon/utils/FutureUtils.scala (hunk 2 above)
Comment (nukky94) [reply src/main/scala/co/ledger/wallet/daemon/utils/FutureUtils.scala]: I see, what do you propose? `failFast` ?

File: src/main/scala/co/ledger/wallet/daemon/services/PoolsService.scala (hunk 1 above)
Comment (luchenyuxx) [reply src/main/scala/co/ledger/wallet/daemon/services/PoolsService.scala]: I mean the actual sync won't happen in global ec even if you remove this ec.
They are happening in the dedicated lib core ec.
So what your wrote is not necessary. You can just remove it.

File: src/main/scala/co/ledger/wallet/daemon/utils/FutureUtils.scala (hunk 2 above)
Comment (luchenyuxx) [reply src/main/scala/co/ledger/wallet/daemon/utils/FutureUtils.scala]: No, it's fine.

**Pull request closed without merging**